src/main/java/com/aws/greengrass/componentmanager/ClientConfigurationUtils.java

```diff
@@ -110,23 +120,37 @@ private static void configureClientMutualTLS(ApacheHttpClient.Builder httpBuilde
         }
     }
 
-    private static KeyManager[] createKeyManagers(String privateKeyPath, String certificatePath)
-            throws TLSAuthException {
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.PreserveStackTrace"})
+    KeyManager[] createKeyManagers(String privateKeyPath, String certificatePath) throws TLSAuthException {
         try {
-            List<X509Certificate> certificateChain = EncryptionUtils.loadX509Certificates(Paths.get(certificatePath));
-
-            PrivateKey privateKey = EncryptionUtils.loadPrivateKey(Paths.get(privateKeyPath));
-
-            KeyStore keyStore = KeyStore.getInstance("PKCS12");
-            keyStore.load(null);
-            keyStore.setKeyEntry("private-key", privateKey, null, certificateChain.toArray(new Certificate[0]));
+            return RetryUtils.runWithRetry(SECURITY_SERVICE_RETRY_CONFIG, () -> securityService
+                            .getKeyManagers(compatibleUriString(privateKeyPath), compatibleUriString(certificatePath)),
+                    "get-key-managers", logger);
+        } catch (InterruptedException e) {
+            logger.atError().setCause(e).kv("privateKeyPath", privateKeyPath).kv("certificatePath", certificatePath)
+                    .log("Got interrupted during getting key managers for TLS handshake");
+            throw new TLSAuthException("Get key managers interrupted");
+        } catch (Exception e) {
+            logger.atError().setCause(e).kv("privateKeyPath", privateKeyPath).kv("certificatePath", certificatePath)
+                    .log("Error during getting key managers for TLS handshake");
+            throw new TLSAuthException("Error during getting key managers", e);
+        }
+    }
 
-            KeyManagerFactory keyManagerFactory =
-                    KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
-            keyManagerFactory.init(keyStore, null);
-            return keyManagerFactory.getKeyManagers();
-        } catch (GeneralSecurityException | IOException e) {
-            throw new TLSAuthException("Failed to get key manager", e);
+    private String compatibleUriString(String path) throws URISyntaxException {
+        URI u;
+        try {
+            u = new URI(path);
+        } catch (URISyntaxException e) {
+            // if can't parse the path string as URI, try it as Path and use URI default provider "file"
+            logger.atDebug().setCause(e)
+                    .kv("path", path).log("can't parse path string as URI");
+            u = Paths.get(path).toUri();
+        }
+        if (Utils.isEmpty(u.getScheme())) {
```

Comment:
Which case would hit this check? Would Paths.get(path).toUri() not add a valid scheme.

Reply:
The current UNIX/LINUX path would hit this check. The path looks like `/path/to/file`, which will be treated as a valid URI. But it's lack of `scheme` part.
On the other hand, windows path looks like `C:\path\to\file`, which will be parsed as valid URI because of backslash. In this case it hits the path of converting path to URI - `Path.toUri`, which will add `file` scheme.

Reply:
Oh. so new URI() could return a URI with empty scheme? Then should line 150 be inside try to make the obvious?

Reply:
Yeah, it doesn't hurt. Will do.